src/stats/rpc_stats.cpp

```diff
@@ -0,0 +1,82 @@
+            "getmempoolstats\n"
+            "\nReturns the collected mempool statictics.\n"
+            "\nThe samples are segregated in multiple percision groups.\n"
+            "\nSamples time interval is not guaranteed to be constant, hence there\n"
+            "is a time delta in each sample relative to the last sample.\n"
+            "\nResult:\n"
+            "  [\n"
+            "    {\n"
+            "      \"sample_interval\"    : \"interval\",      (numeric) Interval target in seconds\n"
+            "      \"time_from\"          : \"timestamp\",     (numeric) Timestamp, first sample\n"
+            "      \"samples\"            : [\n"
+            "                                 [<delta_in_secs>,<tx_count>,<dynamic_mem_usage>,<min_fee_per_k>],\n"
+            "                                 [<delta_in_secs>,<tx_count>,<dynamic_mem_usage>,<min_fee_per_k>],\n"
+            "                                 ...\n"
+            "                             ]\n"
+            "    }\n"
+            "    ,...\n"
+            "  ]\n"
+            "\nExamples:\n" +
+            HelpExampleCli("getmempoolstats", "") + HelpExampleRpc("getmempoolstats", ""));
+
+    // get stats from the core stats model
+    uint64_t timeFrom = 0;
+    std::vector<unsigned int> groups = CStats::DefaultStats()->mempoolCollector->getPercisionGroupsAndIntervals();
+
+    UniValue groupsUni(UniValue::VARR);
+    for (unsigned int i = 0; i < groups.size(); i++) {
+
+        MempoolSamplesVector samples = CStats::DefaultStats()->mempoolCollector->getSamplesForPercision(i, timeFrom);
+
+        // use "flat" json encoding for performance reasons
+        UniValue samplesObj(UniValue::VARR);
+        for (const struct CStatsMempoolSample& sample : samples) {
+            UniValue singleSample(UniValue::VARR);
+            singleSample.push_back(UniValue((uint64_t)sample.timeDelta));
+            singleSample.push_back(UniValue((uint64_t)sample.txCount));
+            singleSample.push_back(UniValue((uint64_t)sample.dynMemUsage));
+            singleSample.push_back(UniValue(sample.minFeePerK));
+            samplesObj.push_back(singleSample);
+        }
+
+        UniValue sampleGroup(UniValue::VOBJ);
+        sampleGroup.push_back(Pair("percision_interval", (int)groups[i]));
+        sampleGroup.push_back(Pair("time_from", timeFrom));
+        sampleGroup.push_back(Pair("samples", samplesObj));
+
+        groupsUni.push_back(sampleGroup);
+    }
+
+    return groupsUni;
+}
+
+static const CRPCCommand commands[] =
+{
+    //  category              name                      actor (function)         okSafe argNames
+    //  --------------------- ------------------------  -----------------------  ------ ----------
+        {"stats",             "getmempoolstats",        &getmempoolstats,        true,  {}},
+
+};
+
+void RegisterStatsRPCCommands(CRPCTable& tableRPC)
+{
+    for (unsigned int vcidx = 0; vcidx < ARRAYLEN(commands); vcidx++)
```

Comment:
I think we better fix this together with all other `Register*RPCCommands` calls.